draft-fett-oauth-sd-jwt-vc-types.md

```diff
@@ -402,25 +406,91 @@ MUST contain at least one of the following properties:
 
+- `path`: An array indicating the claim or claims that are being addressed, as
+  described below. This property is REQUIRED.
 - `display`: An object containing display information for the claim, as
   described in (#ClaimDisplayMetadata). This property is OPTIONAL.
 - `verification`: A string indicating how the claim is verified, as described in
   (#ClaimVerificationMetadata). This property is OPTIONAL.
 - `sd`: A string indicating whether the claim is selectively disclosable, as
   described in (#ClaimSelectiveDisclosureMetadata). This property is OPTIONAL.
 
-Note: The JSON Pointer MUST point to the respective claim as if all selectively
-disclosable claims were disclosed to the Verifier. That means that a consuming
-application which does not have access to all disclosures may not be able to
-validate the claim.
+## Claim Path {#ClaimPath}
+
+The `path` property MUST be a non-empty array of strings, `null` values, or
+non-negative integers. It is used to select a particular claim in the credential
+or a set of claims. A string indicates that the respective key is to be
+selected, a `null` value indicates that all elements of the currently selected
+array(s) are to be selected, and a non-negative integer indicates that the
+respective index in an array is to be selected.
+
+The following shows a non-normative, simplified example of a credential:
+
+```json
+{
+  "name": "Arthur Dent",
+  "address": {
+    "street_address": "42 Market Street",
+    "city": "Milliways",
+    "postal_code": "12345"
+  },
+  "degrees": [
+    {
+      "type": "Bachelor of Science",
+      "university": "University of Betelgeuse"
+    },
+    {
+      "type": "Master of Science",
+      "university": "University of Betelgeuse"
+    }
+  ],
+  "nationalities": ["British", "Betelgeusian"]
+}
+```
+
+The following shows examples of `path` values and the respective selected
+claims:
+
+- `["name"]`: The claim `name` with the value `Arthur Dent` is selected.
+- `["address", "street_address"]`: The claim `street_address` with the value
+  `42 Market Street` is selected.
+- `["degrees", null, "type"]`: All `type` claims in the `degrees` array are
+  selected.
+- `["nationalities", 1]`: The second nationality is selected.
+
+In detail, the array is processed from left to right as follows:
+
+ 1. Select the root element of the credential, i.e., the top-level JSON object.
+ 2. Process the `path` components from left to right:
+    1. If the `path` component is a string, select the element in the respective
+       key in the currently selected element(s). If any of the currently
+       selected element(s) is not an object, abort processing and return an
+       error. If the key does not exist in any element currently selected,
+       remove that element from the selection.
+    2. If the `path` component is `null`, select all elements of the currently
+       selected array(s). If any of the currently selected element(s) is not an
```

Comment:
The possible plural on "array(s)" here jumped out at me and gave pause. I wonder putting a limit of a max of one `null` element per path array would be a useful simplification (and acceptable limitation)?

Reply:
I think that something like "select the names of the courses required for all degrees earned" or something like that is a valid use case. I'm also not sure whether such a limitation would make the implementation significantly easier.

Reply:
I'm also not sure to be honest but it seemed like potentially useful simplification. The "something like ... or something like" case you have doesn't seem invalid but I just don't know if it's necessary either. I'm honestly not sure though so feel free to resolve this comment having considered and dismissed the idea :)